src/main/java/vic/cinema/dao/impl/OrderDaoImpl.java

```diff
@@ -0,0 +1,54 @@
+package vic.cinema.dao.impl;
+
+import java.util.List;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+import vic.cinema.dao.OrderDao;
+import vic.cinema.exeptions.DataProcessingException;
+import vic.cinema.lib.Dao;
+import vic.cinema.model.Order;
+import vic.cinema.model.User;
+import vic.cinema.util.HibernateUtil;
+
+@Dao
+public class OrderDaoImpl implements OrderDao {
+    @Override
+    public Order create(Order order) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(order);
+            transaction.commit();
+            return order;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Order has not been added \n", e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public List<Order> getAllByUser(User user) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<Order> query = session.createQuery(
+                    "FROM Order o "
+                    + "JOIN FETCH o.tickets "
+                    + "JOIN FETCH o.user "
```

Comment:
Do we need to fetch the user?
Which **_fetch_** type is set by default in the `@ManyToOne` and `@OneToOne` connections?

Reply:
@DariiaPikul @jah701 let's test if the additional request is made to db to eager-fetch user
if yes, then it is nothing wrong with join fetching user

unfortunately, it seems that fetchType.EAGER does not guarantee that all associated entities will be fetched with one request
https://vladmihalcea.com/eager-fetching-is-a-code-smell/

Reply:
Fetch type for @OneToOne is EAGER. But without "JOIN FETCH o.user " I got 3 select-queries in a traice. In my version (with "JOIN FETCH o.user ") - I got 2 select-queries.